File: server_asp/src/models/Promotion.cs
using System;
using System.Collections.Generic;

namespace shieldtify.models;

public partial class Promotion
{
    public string Code { get; set; } = null!;

    public string Description { get; set; } = null!;

    public string Type { get; set; } = null!;

    public string Condition { get; set; } = null!;

    public float DiscountRate { get; set; }

    public float MaxDiscount { get; set; }

    public DateTime StartDate { get; set; }

    public DateTime EndDate { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }

    public virtual ICollection<Order> Orders { get; set; } = new List<Order>();
}

File: server_nodejs/src/models/promotion.js
// Promotion (UID, name, description, discount_rate, start_date, end_date)

module.exports = (sequelize, Sequelize) => {
    class Promotion extends Sequelize.Model {}

    Promotion.init(
        {
            code: {
                type: Sequelize.STRING,
                primaryKey: true,
            },
            description: {
                type: Sequelize.STRING,
                allowNull: false,
            },
            type: {
                //enum: ['by total', 'by item'],
                type: Sequelize.STRING,
                allowNull: false,
            },
            condition: {
                type: Sequelize.TEXT('long'),
                allowNull: false,
            },
            discount_rate: {
                type: Sequelize.FLOAT,
                allowNull: false,
            },
            max_discount: {
                type: Sequelize.FLOAT,
                allowNull: false,
            },
            start_date: {
                type: Sequelize.DATE,
                allowNull: false,
            },
            end_date: {
                type: Sequelize.DATE,
                allowNull: false,
            },
        },
        {
            sequelize,
            modelName: 'promotion',
            timestamps: true,
            paranoid: false,
            underscored: true,
        },
    );

    Promotion.associate = (models) => {
        Promotion.hasMany(models.order, {
            foreignKey: 'promotion_code',
        });
    };

    return Promotion;
};
